### PR TITLE
Improve the rendering of the docs 

### LINKED ISSUE #5159: Improve the rendering of the docs
### This is not a Bug Report, Feature Request, or related to Documentation

- [X] I have searched the existing issues

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What's up?

While writing docs we found some issues: 

1. The performance of loading the docs is a blocker. It could some time take minutes to load. 
2. The `@link` modifier does not work across packages. 
3. The layout of the large number of properties makes them difficult to read 

### Examples/References

_No response_

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Fixes #5159

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` with success.
- [x] I have tested the built `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Comment by nazarhussain** [comment]
Improve the rendering of the docs in more readable structure. 

image

**Comment by nazarhussain** [comment]
It improves the performance from over a minute to seconds. 

```
Debug: Finished getting entry points in 18856ms
Debug: Finished conversion in 2792ms
Debug: JSON rendering took 668ms
```

**Comment by nazarhussain** [comment]
The only issue it has is the referencing the types across the packages, which is [reported](https://github.com/milesj/docusaurus-plugin-typedoc-api/issues/47) to plugin. I am hopeful that can be resolved and we can upgrade the plugin later.

**Review by luu-alex — APPROVED** [review]

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -44,46 +65,23 @@ const config = {
...
-					'../packages/web3-validator',
-				],
-				entryPointStrategy: 'packages',
-				cleanOutputDir: true,
-				sidebar: {
-					fullNames: true,
+				projectRoot: join(__dirname, '..'),
+				// Monorepo
+				packages: packages.map(p => `packages/${p}`),
+				minimal: false,
```
Comment (jdevcs) [inline docs/docusaurus.config.js:73]: Its by default false for rendering minimal layout

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -44,46 +65,23 @@ const config = {
...
+				debug: true,
+				changelogs: true,
+				readmes: false,
```
Comment (jdevcs) [inline docs/docusaurus.config.js:76]: Its also false by default.

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -17,6 +17,27 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
...
+const { join } = require('path');
+
+const packages = [
+	'web3',
+	'web3-eth',
+	'web3-eth-contract',
+	'web3-utils',
+	'web3-validator',
+	'web3-common',
+	'web3-core',
```
Comment (jdevcs) [inline docs/docusaurus.config.js:29]: What you think should we include documenting web3.js internal packages as well.

File: docs/docusaurus.config.js (hunk 1 above)
Comment (nazarhussain) [reply docs/docusaurus.config.js:73]: Yes, but I want to keep it mention explicitly for readability and future safe.

File: docs/docusaurus.config.js (hunk 2 above)
Comment (nazarhussain) [reply docs/docusaurus.config.js:76]: Yes, but I want to keep it mention explicitly for readability and future safe.

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -123,8 +121,7 @@ const config = {
 						position: 'left',
 					},
 					{
-						to: 'docs/api/', // 'api' is the 'out' directory
-						activeBasePath: 'docs',
+						to: 'api', // 'api' is the 'out' directory
 						label: 'API',
 						position: 'left',
 					},
```
Comment (jdevcs) [inline docs/docusaurus.config.js:127]: Its not part of this PR but for now ( until we have main 4.0.0 released and as default branch on repo ) lets make:
`href: 'https://github.com/ChainSafe/web3.js',`
to
`href: 'https://github.com/ChainSafe/web3.js/tree/4.x',`

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: docs/docusaurus.config.js (hunk 3 above)
Comment (nazarhussain) [reply docs/docusaurus.config.js:29]: These are not just internal packages, these are public packages. So had to be included anyway.

Comment (jdevcs) [reply docs/docusaurus.config.js:29]: - These are not directly part of eth api
- also considering 1.x parity docs
- We will not test these packages as standalone for system tests

But we can keep them in doc for users ref, and can document on least priority as compared to eth api packages.

Comment (nazarhussain) [reply docs/docusaurus.config.js:29]: As per the the package definition these are not private packages, so must be published and documented. 

https://github.com/ChainSafe/web3.js/blob/80b07dd2d670749599123e4fdd9617d9ad5d89dd/packages/web3-core/package.json#L2

All of the code in every package is directly inheriting logic form these packages so would not be possible to completely document Eth or other packages without documenting these.

Comment (jdevcs) [reply docs/docusaurus.config.js:29]: For sure these are **not** private packages, and already mentioned:

> But we can keep them in doc for users ref, and can document on least priority as compared to eth api packages.

( after alpha )

**Comment by jdevcs** [comment]
lgtm, only some tests are failing so may be re trigger.

File: docs/docusaurus.config.js (hunk 3 above)
Comment (nazarhussain) [reply docs/docusaurus.config.js:29]: We can't complete the Eth or any other documentation without documenting `web3-core` and `web3-common` packages. e.g. If we need to document property `web3.eth.defaultAccount` that is inherited from the `Web3Context` in `web3-core` package, so it must be documented. Similarly for other components in those shared packages.

**Comment by nazarhussain** [comment]
@jdevcs If the changes looks good to you, then please approve. so I can merge it as soon all tests passes.

**Review by avkos — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

File: docs/docusaurus.config.js (hunk 3 above)
Comment (jdevcs) [reply docs/docusaurus.config.js:29]: These properties are used and exposed via main API packages like eth , contract etc.
